File: LeetCode/0744.Find-Smallest-Letter-Greater-Than-Target.cpp
// 744. Find Smallest Letter Greater Than Target

class Solution {
public:
    char nextGreatestLetter(vector<char>& letters, char target) {
        int n = letters.size();
        int low = 0;
        int high = n-1;
        while(low<=high) {
            int mid = (high-low)/2 + low;
            if(letters[mid]<=target) {
                low = mid+1;
            }
            else {
                high = mid-1;
            }
        }
        if(low==n) return letters[0];
        return letters[low];
    }
};

File: LeetCode/0744-Find-Smallest-Letter-Greater-Than-Target.java
// 744. Find Smallest Letter Greater Than Target

class Solution {
    public char nextGreatestLetter(char[] letters, char target) {
        int n = letters.length;
        int l = 0, h = n - 1;
        char ans = letters[0];
        while (l <= h) {
            int m = (h - l) / 2 + l;
            if (letters[m] > target) {
                ans = letters[m];
                h = m - 1;
            }
            else {
                l = m + 1;
            }
        }
        return ans;
    }
}
